zenoh-ext/examples/z_nbftr_cache.rs

```diff
@@ -0,0 +1,115 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use async_std::task::sleep;
+use clap::{App, Arg};
+use futures::prelude::*;
+use std::time::Duration;
+use zenoh::config::Config;
+use zenoh::prelude::r#async::*;
+use zenoh_ext::*;
+
+#[async_std::main]
+async fn main() {
+    // Initiate logging
+    env_logger::init();
+
+    let (config, key_expr, history, prefix) = parse_args();
+
+    println!("Opening session...");
+    let session = zenoh::open(config).res().await.unwrap().into_arc();
+
+    println!("Declaring NBFTReliabilityCache on {}", key_expr);
+    let _cache = session
+        .declare_reliability_cache(key_expr)
+        .history(history)
```

Comment:
The default history is 1024

Reply:
The size of the reliability cache is really dependent on the use case and should be configured accordingly. This will force the user to think how big or small it has to be. Similar thing applies for the publisher and subscriber. Should we make it mandatory?